packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -568,3 +569,143 @@ class _TextSelectionHandleOverlayState extends State<_TextSelectionHandleOverlay
     return null;
   }
 }
+
+/// A gesture detector to detect concurrent gesture events for a text field.
+///
+/// Unlike a plain [GestureDetector] where a single gesture wins the gesture
+/// arena, text fields supports chains of concurrent events.
+///
+/// See also:
+///
+///  * [TextField], a Material text field which uses this gesture detector.
+///  * [CupertinoTextField], a Cupertino text field which uses this gesture
+///    detector.
+class TextSelectionGestureDetector extends StatefulWidget {
+  /// Create a [TextSelectionGestureDetector].
+  ///
+  /// Multiple callbacks can be called from the same sequence of gesture.
+  /// The [child] parameter must not be null.
+  const TextSelectionGestureDetector({
+    this.onTapDown,
+    this.onSingleTapUp,
+    this.onSingleTapCancel,
+    this.onSingleLongTapDown,
+    this.onDoubleTapDown,
+    @required this.child,
+  }) : assert(child != null);
+
+  /// Called each distinct time a tap was held momentarily or on a short tap.
```

Comment:
Not sure what the difference between "tap was held momentarily" and "a short tap" is.

Reply:
Reworded this part